propagation.go

```diff
@@ -115,9 +115,9 @@ func (p *textMapPropagator) Inject(
 	// Do not encode the string with trace context to avoid accidental double-encoding
 	// if people are using opentracing < 0.10.0. Our colon-separated representation
 	// of the trace context is already safe for HTTP headers.
-	textMapWriter.Set(TracerStateHeaderName, sc.String())
+	textMapWriter.Set(p.tracer.headerKeys.TracerStateHeaderName, sc.String())
```

Comment:
I agree. I don't think these headers should be known to the tracer in the first place, it smells of poor composition, since only the text map codec needs them. Can it be localized to the config struct only? If the custom headers are provided then config can instantiate new codecs and pass them to the tracer constructor to override the default ones.